src/decoder.rs

```diff
@@ -223,10 +195,72 @@ fn decode_load(raw: usize) -> Instr {
     let rd = Register::from(rd);
 
     return match func3 {
-        0b000 => Instr::Lb { rd, rs1, imm },
-        0b001 => Instr::Lh { rd, rs1, imm },
-        0b010 => Instr::Lw { rd, rs1, imm },
-        0b011 => Instr::Ld { rd, rs1, imm },
+        0b000 => Instr::Load {
+            rd,
+            rs1,
+            imm,
+            len: (8),
+            compressed: (false),
+            unsigned: (false),
+        },
+        0b001 => Instr::Load {
+            rd,
+            rs1,
+            imm,
+            len: (16),
+            compressed: (false),
+            unsigned: (false),
+        },
+        0b010 => Instr::Load {
+            rd,
+            rs1,
+            imm,
+            len: (32),
+            compressed: (false),
+            unsigned: (false),
+        },
+        0b011 => Instr::Load {
+            rd,
+            rs1,
+            imm,
+            len: (64),
+            compressed: (false),
+            unsigned: (false),
+        },
+        0b100 => Instr::Load {
+            rd,
+            rs1,
+            imm,
+            len: (8),
+            compressed: (false),
+            unsigned: (true),
+        },
+        0b101 => Instr::Load {
+            rd,
+            rs1,
+            imm,
+            len: (16),
+            compressed: (false),
+            unsigned: (true),
+        },
+        0b110 => Instr::Load {
+            rd,
+            rs1,
+            imm,
+            len: (32),
+            compressed: (false),
+            unsigned: (true),
+        },
+        // On a RV64 ISA this should never be the case,
+        // But for the sake of completeness we might decode it as well
```

Comment:
Which instruction is that? 128 bits load?
If it is we should return `Instr::Unknown`, otherwise we will silently emulate it and not be compliant with the spec.

Reply:
It would be an LDU instruction, it's a part of RV128I set.
Sure, i'll remove that match option.